spec/services/expression_evaluators/measure_unit_spec.rb

```diff
@@ -32,43 +32,87 @@
     )
   end
 
-  let(:expected_evaluation) do
-    {
-      calculation: '35.10 EUR / 100 kg * 120.0',
-      value: 3596.12136,
-      formatted_value: '£3,596.12',
-      total_quantity: 120.0,
-      unit: 'x 100 kg',
-    }
-  end
+  context 'with an integer quantity' do
+    let(:expected_evaluation) do
+      {
+        calculation: '35.10 EUR / 100 kg * 120.0',
+        value: 3596.12136,
+        formatted_value: '£3,596.12',
+        total_quantity: 120.0,
+        unit: 'x 100 kg',
+      }
+    end
 
-  let(:session_attributes) do
-    {
-      'import_date' => '2022-01-01',
-      'customs_value' => {
-        'insurance_cost' => '10',
-        'monetary_value' => '10',
-        'shipping_cost' => '10',
-      },
-      'measure_amount' => {
-        'dtn' => '120',
-      },
-      'commodity_source' => 'xi',
-      'commodity_code' => '0103921100',
-    }
-  end
+    let(:session_attributes) do
+      {
+        'import_date' => '2022-01-01',
+        'customs_value' => {
+          'insurance_cost' => '10',
+          'monetary_value' => '10',
+          'shipping_cost' => '10',
+        },
+        'measure_amount' => {
+          'dtn' => '120',
+        },
+        'commodity_source' => 'xi',
+        'commodity_code' => '0103921100',
+      }
+    end
+
+    let(:user_session) { build(:user_session, session_attributes) }
 
-  let(:user_session) { build(:user_session, session_attributes) }
+    it 'returns a properly calculated evaluation' do
+      expect(evaluator.call).to eq(expected_evaluation)
+    end
 
-  it 'returns a properly calculated evaluation' do
-    expect(evaluator.call).to eq(expected_evaluation)
+    context 'when on a route with applicable deltas' do
+      let(:user_session) { build(:user_session, :deltas_applicable, session_attributes) }
+
+      it 'returns a properly calculated evaluation' do
+        expect(evaluator.call).to eq(expected_evaluation)
+      end
+    end
   end
 
-  context 'when on a route with applicable deltas' do
-    let(:user_session) { build(:user_session, :deltas_applicable, session_attributes) }
+  context 'with a quantity with more four decimals' do
```

Comment:
I think if you delegate to the number_to_precision helper with whatever locale we've setup you don't really need a dedicated unit test for this?